### DISCUSSION TITLE
Quick Access Mod Menu

### DISCUSSION THREAD

**Discussion by cloudiees** [ppy/osu#30701] [Ideas]
**THIS IS JUST INTRO SKIP BELOW FOR THE ACTUAL IDEA**
I find the current mod menu sort of overwhelming. There are a lot of mods in lazer and I've seen people wanting to consolidate mods via mod customization, however there are some mods that would make sense to consolidate but I personally would not like. For example, NC into DT, personally I play NC, I don't want to have to go into the mod customization menu to toggle NC (I'm lazy ik).

Also, I saw a post regarding some sort of indicator for the hotkeys of mods in the mod menu, and it made me think that the current hotkey system for the mod menu just feels unusual (like why do some mods have hotkeys and others don't).

**THE ACTUAL IDEA**
A separate mod menu. This mod menu would be similar to stable's mod menu, where there is a set number of grid spaces, but let's make it a rectangle. These grid spaces would each have an associated hotkey to them. Also, these grid spaces can be layered, meaning that you can have multiple mods on the same grid square and cycle through them, similar to the stable DT/NC or SD/PF mod selection. This menu should be configurable by the user, meaning they can set their menu squares to whatever they want and layer whatever mods they want on the each square, and even change the keybinds associated with each menu square. Also, allowing specific mod customization would also be cool, so you could do things like stack dt(x1.5), dt(x1.75), and dt(x2.0) on one square. The default quick access menu should be similar to the stable mod selection. Then, within this quick access mod menu there should be a button to go to the full mod menu.

I feel like this would make the mod menu significantly less overwhelming, especially if you are a new user and want to play around with mods. I also believe this will make it easier for people to cycle through random mod combos they typically use (ex: going for HR to HRHD or DT to NC<insert random mod>) without having to make a bunch of presets and searching for their desired mods.

Thoughts?

**Comment by N0Y0U2114** [comment]
I think much of this could be done without an entire UI rework, simillar presets auto stack would be nice. I think that having a fully customizable grid would be more of a clutter to a new user compared to the current way it works (Clear labels, mods sorted into pretty understandable catagories). I think being able to custom sort, pin, and stack simillar presets would fix most of the issues
